service-worker.js

```diff
@@ -167,20 +95,14 @@ function fetchEventListener(event) {
                 messageChannel.port1.onmessage = function(event) {
                     if (event.data.action === 'giveContent') {
                         // Content received from app.js
-                        var contentLength;
+                        var contentLength = event.data.content ? event.data.content.byteLength : null;
+                        var contentType = event.data.mimetype;
                         var headers = new Headers ();
-                        if (event.data.content && event.data.content.byteLength) {
-                            contentLength = event.data.content.byteLength;
-                            headers.set('Content-Length', contentLength);
-                        }
-                        if (contentType) {
-                            headers.set('Content-Type', contentType);
-                        }
+                        if (contentLength) headers.set('Content-Length', contentLength);
+                        if (contentType) headers.set('Content-Type', contentType);
                         // Test if the content is a video.
-                        // String.prototype.startsWith is not supported by IE11, but IE11 does not support service workers, so it's safe to use it here.
-                        // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith
-                        if (contentLength && contentLength >= 1 && contentType && contentType.startsWith('video/')) {
-                            // In case of a video, Chrome and Edge need these HTTP headers else seeking doesn't work
+                        if (contentLength >= 1 && /video|audio|mp4|webm|og[gmv]|mpeg/i.test(contentType)) {
```

Comment:
It would be good to add a comment here to explain why we test for mp4, webm, og*, and mpeg, even if they should be matched by video, with a link to the issues you created for the invalid mime-types.
It would also be better to be more specific in the regexp, to match only the content types that _start_ with one of these strings

Reply:
OK, I've made the regex more specific, and have tested it carefully in regexbuddy for all possible combinations. It will only match primary types (audio, video) at the beginning of the string, and subtypes either at the beginning of the string or after an /, and subtypes must also be at the end of the string. Have added comment.

Reply:
I think we should not match subtypes beginning with /.
IMHO, we should only match : 
- the regular audio and video mime-types (with ^video or ^audio)
- the known unregular mime-types (^mp4$, ^webm$, ^og[gmv]$, ^mpeg$)

It should not match something like application/mp4encoder (this probably does not exist)

Reply:
It can't match application/mp4encoder because then mp4 wouldn't be at the end of the string. As I've written the regex, it can **only** match `^mp4$` or `/mp4$`, i.e., mp4 on its own, or mp4 after a forward slash and with no further characters after it. Is that OK?

Reply:
![image](https://user-images.githubusercontent.com/4304337/58976499-81575b80-87bf-11e9-903b-edd2c62c268d.png)

Reply:
You're right.
But why should it match application/ogg, for example? It's not a known exception.
I would have used `^(video|audio)|^(mp4|webm|og[gmv]|mpeg)$`, to match only what is necessary, but it's minor.

Reply:
I know it looks strange, and I queried it with kelson, but it has to match `application/ogg` because that's the MIME type used in zimwriterfs for OGG audio or video. See the code here: https://github.com/openzim/zimwriterfs/blob/master/src/tools.cpp#L75 . If we don't match `/ogg` we wouldn't match this MIME type.

Reply:
OK sorry, I had not seen.
In this case, your regexp is perfectly fine.